students/GuyCChan/session09/test_objOriMailRoom.py

```diff
@@ -0,0 +1,226 @@
+from objOriMailRoom import Donor, DonorCollection, make_report, letters_to_donors, test_data
+import pytest
+
+# Tests for Donor class.
+def test_init():
+    db = donors_db()
+    assert len(donors_db) == 0
+
+def test_Donor_in_db():
+    donor = donor("Mike May")
+    assert donor.name == "Mike May"
+    assert donor.donation is None
+
+def test_add_donation():
+    add_donation = self.donations("100.00")
+    assert add_donation == 100.00
+
+def test_tot_donations():
+    db = donors_db({"Teri Thomas": [60.00, 49,50, 100.50]})
+    assert sum(self.donations) == 210.00
```

Comment:
there is no "self" here -- self is only for within method definitions.

"self" means "the current instance of the class" -- when you are using an instance of the class, then it it referred to by whatever name you gave it.